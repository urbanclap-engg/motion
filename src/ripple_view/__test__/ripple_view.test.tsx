import * as React from 'react';
import { render } from '@testing-library/react-native';
import { View } from 'react-native';
import Ripple from '../ripple_view';

beforeEach(() => {
	jest.resetAllMocks();
});

describe('motion_view', () => {
	it('should render', () => {
		const { getByTestId, toJSON } = render(
			<Ripple>
				<View style={{ width: 100, height: 100, backgroundColor: 'red' }} />
			</Ripple>,
		);
		expect(getByTestId('ripple_view')).toBeTruthy();
		const tree = toJSON();
		expect(tree).toMatchSnapshot();
	});
});
