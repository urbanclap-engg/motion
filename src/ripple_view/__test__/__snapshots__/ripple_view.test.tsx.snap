// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`motion_view should render 1`] = `
<View
  isForced={true}
>
  <View
    style={
      Array [
        undefined,
        Object {
          "height": "auto",
          "overflow": "hidden",
          "width": "auto",
        },
      ]
    }
    testID="ripple_view"
  >
    <View
      collapsable={false}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "red",
            "height": 100,
            "width": 100,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "rgba(0,0,0,0.2)",
            "borderRadius": 0,
            "height": 0,
            "left": 0,
            "opacity": 1,
            "position": "absolute",
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
              Object {
                "scale": 0,
              },
            ],
            "width": 0,
          }
        }
      />
    </View>
  </View>
</View>
`;
